_posts/2024-04-23-secure-blog-migration.md

```diff
@@ -0,0 +1,151 @@
+By resolving this security issue, we seized the opportunity to migrate our blog 
+repository from GitHub Organization to GitHub Enterprise. 
+However, the old repository remains in place to preserve the original 
+domain name and redirect users to our new website.
+
+## Action plan details
+We've used Terraform as Infrastructure as a Code combined with GitHub Actions to deploy all our resources.
+1. AWS WAF
+* Implemented **Rate limiting** to protect against DDoS Attacks.
+* Integrated Managed AWS Rules:
+  * **Common Rules** to protect against exploitation of a wide range of vulnerabilities including those described in OWASP publications.
+  * **Known Bad Input Rules** to block requests patterns that are known to be invalid and are associated with  exploitation or discovery of vulnerabilities.
+  * **Amazon Ip reputation List**  to block sources associated with bots or other threats.
+
+2. AWS S3 Bucket
+* We created an S3 bucket that is fully private and only accessible by Cloudfront. 
+
+3. AWS ACM & Route 53
+* Created a subdomain for our blog using Route 53. 
+* Generated a certificate via AWS ACM to ensure secure connections for our visitors. 
+This certificate will be integrated into our CloudFront configuration.
+
+4. AWS Cloudfront
+* Initiated configuration of the origin, directing to our S3 bucket.
+* Following the setup of CloudFront security, we proceeded to connect our WAF
+* Implemented crucial security header, including:
+  * **Strict-Transport-Security (STS)** to enforce https
+  * **Content-Security-Policy (CSP)** to protect your site from XSS attacks. By whitelisting resources of approved content, you can prevent the browser from loading malicious assets
+  * **X-Frame-Options** to prevent your content from being used in a frame by attackers (clickjacking).
+  * **X-Content-Type-Options** to stop the browser from MIME sniffing and force it to stick with the declared content-type. The only valid value for this header is `X-Content-Type-Options: no-sniff`.
+  * **Referrer-Policy** to stop sending referrer info in the headers. This way we increase the users privacy when redirecting to other sites.
+  * **Permissions-Policy** to control which features (camera, location, mic, ...) and APIs can be used in the browser. 
+  * **X-XSS-Protection** to enable XSS protection on old browsers. Modern browsers uses the CSP headers to do this.
+
+### Final infrastructure
+<img src="{{ '/img/2024-04-23-secure-blog-migration/blog_infra.svg' | prepend: site.baseurl }}" alt="AWS infra" class="image fit" style="margin:0px auto; max-width:100%">
+
+
+## Result
+Through this transition to AWS infrastructure and the implementation of 
+comprehensive security measures, we've significantly enhanced the security 
+and reliability of our blog.
+
+<img src="{{ '/img/2024-04-23-secure-blog-migration/security_report_good.png' | prepend: site.baseurl }}" alt="Good Security Report" class="image fit" style="margin:0px auto; max-width:100%">
+
+
+## Conclusion
```

Comment:
AI generated? It repeats the same point multiple time. Rewrite to make it more fluent and less repetitive.